hyperspy/signal.py

```diff
@@ -1924,8 +1924,18 @@ def _get_undefined_axes_list(self):
     def __call__(self, axes_manager=None):
         if axes_manager is None:
             axes_manager = self.axes_manager
-        return np.atleast_1d(
-            self.data.__getitem__(axes_manager._getitem_tuple))
+        if getattr(self._plot, 'pointer', None) is not None:
+            indices = self._plot.pointer.indices
+            size = self._plot.pointer.get_size_in_indices()
+            array_slices = [slice(None, None, None)] * \
+                self.axes_manager.navigation_dimension
+            for i, axis in enumerate(self.axes_manager.navigation_axes[::-1]):
```

Comment:
Thanks for looking at this!
Do you think is worth trying to add a width` attribute to `DataAxis` and get the slice from the axes directly? For example, in this attempt, when you close the navigation plot and create it again, the pointer is not properly restored because the size of the pointer is not saved.